lib/Backend/arm64/LowerMD.cpp

```diff
@@ -7973,33 +7973,16 @@ void LowererMD::GenerateFloatTest(IR::RegOpnd * opndSrc, IR::Instr * insertInstr
         return;
     }
 
-    if(checkForNullInLoopBody && m_func->IsLoopBody())
-    {
-        // It's possible that the value was determined dead by the jitted function and was not restored. The jitted loop
-        // body may not realize that it's dead and may try to use it. Check for null in loop bodies.
-        //     test src1, src1
-        //     jz $helper (bail out)
-        m_lowerer->InsertCompareBranch(
-            opndSrc,
-            IR::AddrOpnd::NewNull(m_func),
-            Js::OpCode::BrEq_A,
-            labelHelper,
-            insertInstr);
-    }
-
-    IR::RegOpnd *vt = IR::RegOpnd::New(TyMachPtr, this->m_func);
-    IR::Opnd* opnd = IR::IndirOpnd::New(opndSrc, (int32)0, TyMachPtr, this->m_func);
-    LowererMD::CreateAssign(vt, opnd, insertInstr);
-
-    // CMP [number], JavascriptNumber::vtable
-    IR::Instr* instr = IR::Instr::New(Js::OpCode::CMP, this->m_func);
-    instr->SetSrc1(vt);
-    instr->SetSrc2(m_lowerer->LoadVTableValueOpnd(insertInstr, VTableValue::VtableJavascriptNumber));
+    // TST s1, floatTagReg
```

Comment:
This is doing a tagged float check and LoadFloatValue relies on this and loads value from offset which is correct?

Reply:
This is doing a tagged float check, but LoadFloatValue is still ARM implementation, expected non-tagged floats, so that will have to be updated as well.